google/cloud/bigquery/magics/magics.py

```diff
@@ -679,13 +724,18 @@ def _cell_magic(line, query):
             job_config.maximum_bytes_billed = value
 
         try:
-            query_job = _run_query(client, query, job_config=job_config)
+            query_job = _run_query(client, query, args, job_config=job_config)
         except Exception as ex:
             _handle_error(ex, args.destination_var)
             return
 
         if not args.verbose:
-            display.clear_output()
+            clear_output()
+
+        if args.send_long_job:
+            if args.destination_var:
+                IPython.get_ipython().push({args.destination_var: result})
```

Comment:
execution can reach here while job thread is still running in the async case. i think you'd want to refactor this so that there is one copy of the result post-processing code, and it gets called in both scenarios (sync and async) after completion of the job (maybe as a separate function).